src/jsonDataGenerator.js

```diff
@@ -75,7 +79,57 @@ async function prepareTopic(schema, schemaFormat, dryRun) {
     }
 }
 
-module.exports = async ({ schema, records, schemaFormat, dryRun = false }) => {
+function getAvroSchema(topic, record, debug = false){
+    let avro_schema = Type.forValue(record).schema();
```

Comment:
```suggestion
    let avro_schema = Type.forValue(record[Object.keys(record)[0]]).schema();
```

Reply:
This change allows us to pass only the values, eg:

- Instead of:

```
{
  merchants: {
    id: 58556,
    country_code: 2856,
    merchant_name: 'welcome',
    'created at': 'illustrious',
    admin_id: 85626
  }
}
```
- We will pass:
```
{
  id: 58556,
  country_code: 2856,
  merchant_name: 'welcome',
  'created at': 'illustrious',
  admin_id: 85626
}
```

that way the generated schema will be valid

Reply:
When I look in [schema registry](https://confluent.cloud/environments/env-vo365/schema-registry/schemas/mz_datagen_comments-value) for the mz_datagen_comments example, the schema looks correct.

I think there might actually be a bug in the sql parser. In the merchants example, I think merchants should be in the topic name and should not be included in the record at all. What do you think?

Reply:
Oh yes you are correct. Indeed, it is only how we generate the SQL payload. I guess then in this case this PR is all good! We could raise a different issue and a PR to fix the SQL schema parser!